internals/core_models.py

```diff
@@ -193,8 +194,25 @@ def format_for_template(self) -> dict[str, Any]:
 
     if self.is_saved():
       d['id'] = self.key.integer_id()
+
+      # Write a collection of stages and gates associated with the feature,
+      # sorted by type.
+      stages: list[Stage] = Stage.query(
```

Comment:
I'm glad you mentioned this, because it is something I'd like to be more efficient, but also I just moved this method in the Feature refactor PR.

After thinking about it, it's probably better to just remove this logic entirely for now. Technically, Feature entities do not have Stages and Gates associated with them - it should be the FeatureEntry entity. A new function will need to be created for FeatureEntry, and that's where I'll handle this logic. It was needed for handling the POST request for the approvals API, but I've refactored that to only take in the feature ID and gate type and discern the correct gate with that information until the new FeatureEntry converter is implemented.

This is something that will need to be revisited, since so many fields are located on Stage and Gate entities that will need to be queried for to have a complete picture of the feature.

Reply:
Sounds good